src/com/facebook/buck/jvm/groovy/GroovyConfiguredCompilerFactory.java

```diff
@@ -42,8 +45,17 @@ public CompileToJarStepFactory configure(
       ToolchainProvider toolchainProvider) {
     GroovyLibraryDescription.CoreArg groovyArgs = (CoreArg) Objects.requireNonNull(args);
     return new GroovycToJarStepFactory(
-        Objects.requireNonNull(groovyBuckConfig).getGroovyCompiler().get(),
+        groovyBuckConfig.getGroovyc(),
```

Comment:
this can trigger NPE

Reply:
added a precondition check since at this step it needs to be non null.